control-plane/api-gateway/binding/route_binding.go

```diff
@@ -80,6 +86,14 @@ func (r *Binder) bindRoute(route client.Object, boundCount map[gwv1beta1.Section
 			kubernetesNeedsStatusUpdate = true
 		}
 	}
+	// if we're orphaned from this gateway we'll
+	// always need a status update.
+	if len(filteredParents) == 0 {
+		// we already checked that these refs existed, so no need to check
+		// the return value here.
+		_ = r.statusSetter.removeRouteReferences(route, filteredParentStatuses)
+		kubernetesNeedsStatusUpdate = true
+	}
```

Comment:
this goes along with the addition above --  basically if we no longer reference the gateway but do so in our statuses, clean up all of said statuses